packages/components/src/internal/components/chart/ChartFieldOption.tsx

```diff
@@ -0,0 +1,230 @@
+                col.dimension
+            );
+            return hasMatchingType || isMeasureDimensionMatch;
+        })
+        .sort(naturalSortByProperty('caption'))
+        .map(col => ({ label: col.caption, value: col.fieldKey, data: col }));
+};
+
+export const shouldShowFieldOptions = (field: ChartFieldInfo, selectedType: ChartTypeInfo): boolean => {
+    const showForX = field.name === 'x' && (selectedType.name === 'scatter_plot' || selectedType.name === 'line_plot');
+    const showForY =
+        field.name === 'y' &&
+        (selectedType.name === 'scatter_plot' || selectedType.name === 'line_plot' || selectedType.name === 'box_plot');
+    return showForX || showForY;
+};
+
+const SCALE_TRANS_TYPES = [
+    { value: 'linear', label: 'Linear' },
+    { value: 'log', label: 'Log' },
+];
+
+const SCALE_RANGE_TYPES = [
+    { value: 'automatic', label: 'Automatic' },
+    { value: 'manual', label: 'Manual' },
+];
+
+const DEFAULT_SCALE_VALUES = { type: 'automatic', trans: 'linear' };
+
+interface ChartFieldOptionProps {
+    field: ChartFieldInfo;
+    fieldValue?: SelectInputOption;
+    model: QueryModel;
+    onScaleChange: (field: string, key: string, value: string | number, reset?: boolean) => void;
+    onSelectFieldChange: (name: string, value: any, selectedOption: SelectInputOption) => void;
+    scaleValues?: Record<string, string | number>;
+    selectedType: ChartTypeInfo;
+}
+
+export const ChartFieldOption: FC<ChartFieldOptionProps> = memo(props => {
+    const { field, model, selectedType, onSelectFieldChange, scaleValues, fieldValue, onScaleChange } = props;
+    const options = useMemo(() => getSelectOptions(model, selectedType, field), [model, selectedType, field]);
+    const isNumericType = useMemo(
+        () => LABKEY_VIS.GenericChartHelper.isNumericType(getFieldDataType(fieldValue?.data)),
+        [fieldValue?.data]
+    );
+    const showFieldOptions = isNumericType && shouldShowFieldOptions(field, selectedType);
+    const [scale, setScale] = useState<Record<string, string | number>>(scaleValues ?? {});
+    const invalidRange = useMemo(
+        () =>
+            scale.min !== undefined &&
+            scale.min !== null &&
+            scale.max !== undefined &&
+            scale.max !== null &&
+            parseFloat(scale.max.toString()) <= parseFloat(scale.min.toString()),
+        [scale]
+    );
+
+    useEffect(() => {
+        if (showFieldOptions && !scale.type) {
+            setScale(scaleValues?.type ? scaleValues : DEFAULT_SCALE_VALUES);
+        }
+    }, [showFieldOptions, scale.type, scaleValues]);
+
+    const onScaleTransChange = useCallback(
+        (selected: string) => {
+            onScaleChange(field.name, 'trans', selected);
+            setScale(prev => ({ ...prev, trans: selected }));
+        },
+        [field.name, onScaleChange]
+    );
+
+    const onSelectFieldChange_ = useCallback(
+        (name: string, value: any, selectedOption: SelectInputOption) => {
```

Comment:
I think it is safe to type `value` as `string` here because `getSelectOptions` is using fieldKey as the value.